DhawaAlkhreej/DhawaAlkhreej/Modules/Calculator/CalculatorVC.swift

```diff
@@ -91,6 +91,43 @@ class CalculatorVC: BaseViewController {
         accomplishedStepExamTF.tintColor = UIColor.primary
     }
     
+    func saveTextfieldData(textFieldName: textFieldName, data: String) {
+        
+        var dic = UserDefaults.standard.value(forKey: K.UserDefaults.savedTextfields) as? [String: String]
+        
+        if dic == nil {
+            var newDic: [String: String]? = [:]
+            newDic?[textFieldName.rawValue] = data
+            UserDefaults.standard.set(newDic, forKey: K.UserDefaults.savedTextfields)
+        } else {
+            dic?[textFieldName.rawValue] = data
+            UserDefaults.standard.set(dic, forKey: K.UserDefaults.savedTextfields)
+        }
+        
+    }
+    
+    private func retrieveTextfieldsData() {
+        
+        if let data = UserDefaults.standard.value(forKey: K.UserDefaults.savedTextfields) as? [String: String] {
+            
+            requestedThanawiyahTF.text = data[textFieldName.requestedThanawiyahTF.rawValue]
+            requestedQuodratTF.text = data[textFieldName.requestedQuodratTF.rawValue]
+            requestedTahsilyTF.text = data[textFieldName.requestedTahsilyTF.rawValue]
+            requestedStepExamTF.text = data[textFieldName.requestedStepExamTF.rawValue]
+            accoplishedThanawiyahTF.text = data[textFieldName.accoplishedThanawiyahTF.rawValue]
+            accomplishedQuodratTF.text = data[textFieldName.accomplishedQuodratTF.rawValue]
+            accomplishedTahsilyTF.text = data[textFieldName.accomplishedTahsilyTF.rawValue]
+            accomplishedStepExamTF.text = data[textFieldName.accomplishedStepExamTF.rawValue]
+        }
+
+        DispatchQueue.main.async {
+            self.validateTextfieldInput(self.accoplishedThanawiyahTF)
```

Comment:
Could you move this to a new method as it's better to have one job only for every method